### PR TITLE
Feat: add Gerrit revision service to Gerrit Instance Handler

### PR DESCRIPTION
**Pull request by timwangmusic**
* This service is a wrapper for APIs on revision endpoints
* Adding one method for getting `MergeableInfo` so that Prow can use the merge strategy instead of always use `Merge if Necessary` on all CLs.
/cc: @listx @cjwagner @mpherman2

### CODE REVIEW

File: prow/gerrit/client/client.go [hunk 1]
```diff
@@ -99,15 +99,20 @@ type gerritProjects interface {
 	GetBranch(projectName, branchID string) (*gerrit.BranchInfo, *gerrit.Response, error)
 }
 
+type gerriRevision interface {
```
Comment (petr-muller) [inline prow/gerrit/client/client.go:102]: `gerri` should be `gerrit`

**Review by petr-muller — COMMENTED** [review]
I'll leave this for the folks who are familiar with gerrit (I'm not) but I noticed a typo

File: prow/gerrit/client/client.go (hunk 1 above)
Comment (timwangmusic) [reply prow/gerrit/client/client.go]: Fixed, thanks Petr.

**Pull request merged**